src/scaling.jl

```diff
@@ -0,0 +1,192 @@
+"""
+    Scaling <: Transform
+
+Linearly scale the data as `ax + b`, according to some statistics `a` and `b`.
+"""
+abstract type Scaling <: Transform end
+
+"""
+    IdentityScaling
+
+Represents the no-op scaling which simply returns the data it is applied on.
+"""
+struct IdentityScaling <: Scaling end
+# Convenience method
+IdentityScaling(args...; kwargs...) = IdentityScaling()
+
+apply!(A::AbstractArray{T}, scaling::IdentityScaling; kwargs...) where T <: Real = A
+
+"""
+    MeanStdScaling(mean, std) <: Scaling
+
+Linearly scale the data by a statistical `mean` and standard deviation `std`.
+The first `mean` and `std` calculated on some data will remain fixed.
+"""
+struct MeanStdScaling <: Scaling
+    # Dict keeps track of slices to protect against changes e.g. Table column order
+    # Is this going too far to generalise between Arrays and Tables? Should we use
+    # separate/parameterised structs?
+    mean::Dict{Symbol, AbstractFloat}
+    std::Dict{Symbol, AbstractFloat}
+
+    """
+        MeanStdScaling() <: Scaling
+
+    Construct a `MeanStdScaling` transform which will populate its `mean` and `std`
+    parameters from the first data it is applied to.
+    """
+    MeanStdScaling() = new(Dict(), Dict())
+end
+
+"""
+    MeanStdScaling(data; kwargs...) <: Scaling
+
+Construct a [`MeanStdScaling`](@ref) using the mean and standard deviation of the data.
+
+# Keyword arguments
+*
+"""
+function MeanStdScaling(data; kwargs...)
+    scaling = MeanStdScaling()
+    populate_stats!(scaling, data; kwargs...)
+    return scaling
+end
+
+function populate_stats!(scaling::MeanStdScaling, A::AbstractArray; dims=:)
+    if dims == Colon()
+        scaling.mean[:all] = mean(A)
+        scaling.std[:all] = std(A)
+    else
+        for (i, x) in enumerate(eachslice(A; dims=dims))
+            scaling.mean[Symbol(i)] = mean(x)
+            scaling.std[Symbol(i)] = std(x)
+        end
+    end
+
+    return scaling
+end
+
+function populate_stats!(scaling::MeanStdScaling, table; cols=nothing)
+    columntable = Tables.columns(table)
+    cnames = cols === nothing ? propertynames(columntable) : cols
+    for cname in cnames
+        x = getproperty(columntable, cname)
+        scaling.mean[cname] = mean(x)
+        scaling.std[cname] = std(x)
+    end
+end
+
+"""
+    apply!(A::AbstractArray, ::Transform; dims=:, kwargs...)
+
+Applies [`MeanStdScaling`](@ref) to each element of `A`.
+Optionally specify the `dims` to apply [`MeanStdScaling`](@ref) along certain dimensions,
+and `inverse=true` to reconstruct the originally scaled data from `A`.
+For `inverse=false`, all 0 std values get replaced by the value `eps` before scaling.
+"""
+function apply!(
+    A::AbstractArray, scaling::MeanStdScaling;
+    dims=:, inverse=false, eps=1e-3, kwargs...
+)
+    if isempty(scaling.mean) || isempty(scaling.std)
```

Comment:
```suggestion
    if isempty(scaling.mean) && isempty(scaling.std)
```

Ideally you want them both to be empty.
Have a separate check that errors if only one of them is populated? Since it's ambiguous what to do in that case.